dashboard/src/components/AppView/AppControls/RollbackButton/RollbackButton.tsx

```diff
@@ -0,0 +1,157 @@
+import * as React from "react";
+import * as Modal from "react-modal";
+import { IAppRepository, IChartVersion, IRelease } from "shared/types";
+import SelectRepoForm from "../../../../components/SelectRepoForm";
+import RollbackDialog from "./RollbackDialog";
+
+interface IRollbackButtonProps {
+  app: IRelease;
+  rollbackApp: (
+    chartVersion: IChartVersion,
+    releaseName: string,
+    revision: number,
+    namespace: string,
+    values?: string,
+  ) => Promise<boolean>;
+  chart?: IChartVersion;
+  getChartVersion: (id: string, version: string) => Promise<void>;
+  loading: boolean;
+  repos: IAppRepository[];
+  repo: IAppRepository;
+  kubeappsNamespace: string;
+  fetchRepositories: () => void;
+  checkChart: (repo: string, chartName: string) => Promise<boolean>;
+  error?: Error;
+}
+
+interface IRollbackButtonState {
+  modalIsOpen: boolean;
+  loading: boolean;
+  chartName: string;
+  chartVersion: string;
+}
+
+class RollbackButton extends React.Component<IRollbackButtonProps> {
+  public state: IRollbackButtonState = {
+    modalIsOpen: false,
+    loading: false,
+    chartName: "",
+    chartVersion: "",
+  };
+
+  public componentWillReceiveProps(nextProps: IRollbackButtonProps) {
+    // Store the chart name and version in the state for convenience
+    if (nextProps.app && !this.state.chartName && !this.state.chartVersion) {
+      if (
+        nextProps.app.chart &&
+        nextProps.app.chart.metadata &&
+        nextProps.app.chart.metadata.name &&
+        nextProps.app.chart.metadata.version
+      ) {
+        this.setState({
+          chartName: nextProps.app.chart.metadata.name,
+          chartVersion: nextProps.app.chart.metadata.version,
+        });
+      } else {
+        // This should not be reached, unexpected error
+        throw new Error("The current app is missing its chart information");
+      }
+    }
+  }
+
+  public render() {
+    return (
+      <React.Fragment>
+        <Modal
+          style={{
```

Comment:
Are these styles common across all the modals in the application? If it's the case, I believe it's better to create a common `Modal`. This can be done in a separate PR